packages/labextension/src/index.ts

```diff
@@ -2,34 +2,80 @@ import {
   JupyterFrontEnd,
   JupyterFrontEndPlugin,
 } from '@jupyterlab/application';
+import { INotebookTracker } from '@jupyterlab/notebook';
+import { LabIcon } from '@jupyterlab/ui-components';
+import { ICommandPalette } from '@jupyterlab/apputils';
+import { nullTranslator } from '@jupyterlab/translation';
+import { KernelUsagePanel } from './panel';
+import tachometer from '../style/tachometer.svg';
 
 import { IStatusBar } from '@jupyterlab/statusbar';
 
 import { ITranslator } from '@jupyterlab/translation';
 
 import { MemoryUsage } from './memoryUsage';
 
+namespace CommandIDs {
+  export const getKernelUsage = 'kernel-usage:get';
+}
+
 /**
  * Initialization data for the jupyter-resource-usage extension.
  */
 const extension: JupyterFrontEndPlugin<void> = {
-  id: '@jupyter-server/resource-usage:memory-status-item',
+  id: '@jupyter-server/resource-usage:memory-kernel-status-item',
   autoStart: true,
-  requires: [IStatusBar, ITranslator],
   activate: (
     app: JupyterFrontEnd,
-    statusBar: IStatusBar,
-    translator: ITranslator
+    statusBar: IStatusBar | null,
```

Comment:
We still need these to be listed (in order) in the `optional` list. Otherwise the approach looks good I think.